x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_sync_rules_table_saved_state.test.tsx

```diff
@@ -0,0 +1,253 @@
+        },
+        { sort: { field: 'name' } }
+      );
+    });
+
+    it('syncs only the sorting order', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          sortingOptions: { ...defaultState.sortingOptions, order: 'asc' },
+        },
+        { sort: { order: 'asc' } }
+      );
+    });
+
+    it('syncs only the page number', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          pagination: {
+            ...defaultState.pagination,
+            page: 10,
+          },
+        },
+        { page: 10 }
+      );
+    });
+
+    it('syncs only the page size', () => {
+      expectUrlSync(
+        {
+          ...defaultState,
+          pagination: {
+            ...defaultState.pagination,
+            perPage: 10,
+          },
+        },
+        { perPage: 10 }
+      );
+    });
+  });
+
+  describe('with the storage', () => {
+    it('syncs only the search term', () => {
+      expectStorageSync(
+        { ...defaultState, filterOptions: { ...defaultState.filterOptions, filter: 'test' } },
+        { searchTerm: 'test' }
+      );
+    });
+
+    it('syncs only the show elastic rules filter', () => {
+      expectStorageSync(
+        {
+          ...defaultState,
+          filterOptions: { ...defaultState.filterOptions, showElasticRules: true },
+        },
+        { showCustomRules: false }
+      );
+    });
+
+    it('syncs only the show custom rules filter', () => {
+      expectStorageSync(
+        {
+          ...defaultState,
+          filterOptions: { ...defaultState.filterOptions, showCustomRules: true },
+        },
+        { showCustomRules: true }
+      );
+    });
+
+    it('syncs only the tags', () => {
+      expectStorageSync(
+        {
+          ...defaultState,
+          filterOptions: { ...defaultState.filterOptions, tags: ['test'] },
+        },
+        { tags: ['test'] }
+      );
+    });
+
+    it('syncs only the sorting field', () => {
+      expectStorageSync(
+        {
+          ...defaultState,
+          sortingOptions: { ...defaultState.sortingOptions, field: 'name' },
+        },
+        { sort: { field: 'name' } }
+      );
+    });
+
+    it('syncs only the sorting order', () => {
+      expectStorageSync(
+        {
+          ...defaultState,
+          sortingOptions: { ...defaultState.sortingOptions, order: 'asc' },
+        },
+        { sort: { order: 'asc' } }
+      );
+    });
+
+    it('syncs only the page number', () => {
+      expectStorageSync(
+        {
+          ...defaultState,
+          pagination: {
+            ...defaultState.pagination,
+            page: 10,
+          },
+        },
+        { page: 10 }
+      );
+    });
+
+    it('syncs only the page size', () => {
+      expectStorageSync(
+        {
+          ...defaultState,
+          pagination: {
+            ...defaultState.pagination,
+            perPage: 10,
+          },
+        },
+        { perPage: 10 }
+      );
+    });
+  });
+});
```

Comment:
Same here: at least one case to test that both the URL and storage are updated at the same time when state updated?

Reply:
Good point, I've added such a test.